packages/@stimulus/core/src/target_properties.ts

```diff
@@ -0,0 +1,47 @@
+export function defineTargetProperties(constructor: Function) {
+  const prototype = constructor.prototype
+  const targetNames = getTargetNamesForConstructor(constructor)
+  targetNames.forEach(name => defineLinkedProperties(prototype, {
+    [`${name}Target`]: {
+      get() {
+        return this.targets.find(name)
+      }
+    },
+    [`${name}Targets`]: {
+      get() {
+        return this.targets.findAll(name)
+      }
+    }
+  }))
+}
+
+function getTargetNamesForConstructor(constructor: Function) {
+  const ancestors = getAncestorsForConstructor(constructor)
+  return Array.from(ancestors.reduce((targetNames, constructor) => {
+    getOwnTargetNamesForConstructor(constructor).forEach(name => targetNames.add(name))
+    return targetNames
+  }, new Set as Set<string>))
+}
+
+function getAncestorsForConstructor(constructor: Function) {
+  const ancestors: Function[] = []
+  while (constructor) {
+    ancestors.push(constructor)
+    constructor = Object.getPrototypeOf(constructor)
+  }
+  return ancestors
+}
+
+function getOwnTargetNamesForConstructor(constructor: Function) {
+  const definition = constructor["targets"]
+  return Array.isArray(definition) ? definition : []
+}
+
+function defineLinkedProperties(object: any, properties: PropertyDescriptorMap) {
+  Object.keys(properties).forEach((name) => {
+    if (object[name] === undefined) {
```

Comment:
This will call a `get`er method if defined, which we probably don't want to do.

<img width="260" alt="screen shot 2018-01-22 at 5 43 25 pm" src="https://user-images.githubusercontent.com/5355/35248422-00a4486a-ff9c-11e7-9f2a-059bd2b75006.png">

It's the reason I went with `name in object` in the current implementation: https://github.com/stimulusjs/stimulus/blob/8c141386085d61e4c42593b6b72aa5f8ff7723ed/packages/%40stimulus/core/src/definition.ts#L41

Reply:
👍 99981ec